doc/api/util.md

```diff
@@ -2060,6 +2074,151 @@ console.log(values, positionals);
+general help text is provided, a help option is automatically added unless disabled or already present.
+
+#### Simple usage
+
+By default, providing general help text automatically adds a help option (`-h, --help`) and
+returns formatted help text in `result.helpText`:
+
+```mjs
+import { parseArgs } from 'node:util';
+
+const options = {
+  foo: {
+    type: 'boolean',
+    short: 'f',
+    help: 'use the foo filter',
+  },
+  bar: {
+    type: 'string',
+    help: 'use the specified bar filter',
+  },
+};
+
+const result = parseArgs({
+  help: 'utility to control filters',
+  options,
+});
+
+if (result.values.help) {
+  console.log(result.helpText);
+  // Prints:
+  // utility to control filters
+  // -f, --foo                     use the foo filter
+  // --bar <arg>                   use the specified bar filter
+  // -h, --help                    Show help
+}
+```
+
+#### Custom help option
+
+You can override the auto-added help option by defining your own:
+
+```mjs
+import { parseArgs } from 'node:util';
+
+const options = {
+  foo: {
+    type: 'boolean',
+    short: 'f',
+    help: 'use the foo filter',
+  },
+  bar: {
+    type: 'string',
+    help: 'use the specified bar filter',
+  },
+  help: {
+    type: 'boolean',
+    short: '?',
+    help: 'display help',
+  },
+};
+
+const result = parseArgs({
+  help: 'utility to control filters',
+  options,
+});
+
+if (result.values.help) {
+  console.log(result.helpText);
+  // Prints:
+  // utility to control filters
+  // -f, --foo                     use the foo filter
+  // --bar <arg>                   use the specified bar filter
+  // -?, --help                    display help
+}
+```
+
+#### Advanced configuration
+
+Use `addHelpOption` and `returnHelpText` to customize help behavior:
+
+```mjs
+import { parseArgs } from 'node:util';
+
+const options = {
+  foo: {
+    type: 'boolean',
+    short: 'f',
+    help: 'use the foo filter',
+  },
+  assist: {
+    type: 'boolean',
+    short: '?',
+    help: 'display help',
+  },
+};
+
+const result = parseArgs({
+  addHelpOption: false, // Suppress auto help option
+  returnHelpText: true, // Generate help text anyway
+  options,
+});
+
+if (result.values.assist) {
+  console.log(result.helpText);
+  // Prints:
+  // -f, --foo                     use the foo filter
+  // -?, --assist                  display help
+}
+```
+
+#### Deferred help generation
+
+For performance, you can postpone help text generation until needed:
+
+```mjs
+import { parseArgs } from 'node:util';
+
+const options = {
+  foo: {
+    type: 'boolean',
+    short: 'f',
+    help: 'use the foo filter',
+  },
+};
+
+// First parse without generating help text
+const result = parseArgs({
+  addHelpOption: true,
+  returnHelpText: false,
+  options,
+});
+
+if (result.values.help) {
+  // Generate help text only when needed
+  const { helpText } = parseArgs({
+    help: 'utility to control filters',
+    options,
+  });
+  console.log(helpText);
+}
```

Comment:
parsing the args twice seems quite wasteful to me. If this is a use case that's important, then let's just *always* return a function that lazily computes and returns the help text.

Reply:
I do not think it is an important use case. I was playing with the idea of deferring formatting without a function, following up on a comment you made earlier. Not very successfully. 🤣 

> I personally think a function is better, because then it allows the implementation to defer formatting of the text if it wishes to.